### PR TITLE
Api 24826 section 1 validation

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

Adds validation method and error handling for country
Adds testing for the following 526 request properties in our POST /526 and /526/validate v2 endpoints

- country
- telephone
- zipFirstFive

## Related issue(s)
[API-24826](https://vajira.max.gov/browse/API-24826)

## Testing done

- Postman
- rspec

## What areas of the site does it impact?
modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb
modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb
modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb [hunk 1]
```diff
@@ -25,6 +29,21 @@ def validate_form_526_claimant_certification!
         raise ::Common::Exceptions::InvalidFieldValue.new('claimantCertification',
                                                           form_attributes['claimantCertification'])
       end
+
```
Comment (stiehlrod) [inline modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb:32]: I know this is the pattern in v1, but do we need a method to call a method? Could we remove lines 33-35, and put 'palidate_form_526_current_mailing_address_country!' in the main method `validate_form_526_submission_values!`?

File: modules/claims_api/app/swagger/claims_api/v2/production/swagger.json [hunk 2]
```diff
@@ -1904,8 +1904,8 @@
                     "id": "1",
```
Comment (stiehlrod) [inline modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:1904]: Please remove this file: production swagger.

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb [hunk 3]
```diff
@@ -301,6 +301,96 @@
...
+                json = JSON.parse(data)
+                json['data']['attributes']['veteranIdentification']['mailingAddress']['zipFirstFive'] = zip_first_five
+                data = json.to_json
+                post path, params: data, headers: headers.merge(auth_header)
+                expect(response).to have_http_status(:unprocessable_entity)
+              end
+            end
+          end
+        end
+      end
```
Comment (stiehlrod) [inline modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:393]: a few of the attributes do not have tests, do you mind adding some for them? apartmentOrUnitNumber, numberAndStreet, city, state, vaFileNumber, currentlyVaEmployee, serviceNumber, agreeToEmailRelatedToClaim, email

**Review by stiehlrod — CHANGES_REQUESTED** [review]

File: modules/claims_api/config/schemas/v2/526.json [hunk 4]
```diff
@@ -69,11 +69,13 @@
                 "properties": {
```
Comment (stiehlrod) [inline modules/claims_api/config/schemas/v2/526.json:69]: Should we add a pattern for the email? Something like `EMAIL_REGEX = '/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/'`

File: modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb (hunk 1 above)
Comment (mchristiansonVA) [reply modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb:32]: All set.

File: modules/claims_api/app/swagger/claims_api/v2/production/swagger.json (hunk 2 above)
Comment (mchristiansonVA) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json]: Removed.

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb (hunk 3 above)
Comment (mchristiansonVA) [reply modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:412]: Added tests for all attributes.

File: modules/claims_api/config/schemas/v2/526.json (hunk 4 above)
Comment (mchristiansonVA) [reply modules/claims_api/config/schemas/v2/526.json:70]: Added an email validation pattern and test for same.

**Review by stiehlrod — APPROVED** [review]
LGTM. thanks for the changes!

**Pull request merged**